shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -322,6 +319,12 @@ private void popSystemNavigator() {
     }
   }
 
+  private boolean clipboardHasText() {
+    ClipboardManager clipboard =
+        (ClipboardManager) activity.getSystemService(Context.CLIPBOARD_SERVICE);
+    return clipboard.hasText();
```

Comment:
the documentation says `hasText` is deprecated: https://developer.android.com/reference/android/content/ClipboardManager#hasPrimaryClip()

Reply:
I tried hasPrimaryClip but it completely wasn't working for me in the tests.  It seemed to always return true, even if I setPrimaryClip to null or did clearPrimaryClip.  You can see how I had it before in https://github.com/flutter/engine/pull/24021/commits/72c32b880924365c1ee486561a42ff5664234e1e.  CC @blasten who was helping me with this [on Discord](https://discord.com/channels/608014603317936148/608021010377080866/808487215713878088).  Is that a robolectric problem?